test/e2e/kms/vault_test.go

```diff
@@ -104,6 +106,40 @@ func (suite *VaultSuite) TestSign() {
 	assert.True(suite.T(), ok)
 }
 
+func (suite *VaultSuite) TestPubKeyVerify() {
+	provider := suite.GetProvider("testsign")
+
+	key, err := provider.CreateKey(context.Background())
+	require.Nil(suite.T(), err)
+	require.NotNil(suite.T(), key)
+
+	data := []byte("mydata")
+	sig, signed, err := provider.Sign(context.Background(), data)
+	require.Nil(suite.T(), err)
+	require.NotNil(suite.T(), sig)
+	require.NotNil(suite.T(), signed)
+
+	k, err := provider.PublicKey(context.TODO())
+	require.Nil(suite.T(), err)
+	require.NotNil(suite.T(), key)
+
+	pubKey, ok := k.(*ecdsa.PublicKey)
+	require.True(suite.T(), ok)
+
+	hash := sha256.Sum256(signed)
+	ok = ecdsa.VerifyASN1(pubKey, hash[:], sig)
+	assert.True(suite.T(), ok)
+
+	ok = ecdsa.VerifyASN1(key, hash[:], sig)
+	assert.True(suite.T(), ok)
+
+	verifier := signature.ECDSAVerifier{
+		Key:     pubKey,
+		HashAlg: crypto.SHA256,
+	}
```

Comment:
In theory this is exactly what is happening in `cosign verify`. I wonder if something is getting lost in  translation.